api/tools/src/main/java/org/pmiops/workbench/tools/SetAccessModuleTimestamps.java

```diff
@@ -60,42 +66,54 @@ void applyTimestampUpdateToUser(
       AccessModuleName moduleName,
       @Nullable Timestamp timestamp,
       boolean isBypass) {
-    DbUser dbUser = userDao.findUserByUsername(username);
-    if (dbUser == null) {
-      throw new IllegalArgumentException(String.format("User %s not found!", username));
-    }
-
-    if (moduleName == AccessModuleName.PROFILE_CONFIRMATION && !isBypass) {
-      final String field = isBypass ? "bypass" : "completion";
-      final String time = Optional.ofNullable(timestamp).map(Timestamp::toString).orElse("NULL");
-      LOG.info(
-          String.format(
-              "Updating %s %s time for user %s to %s", moduleName, field, username, time));
-
+    accessModuleService.updateCompletionTime(dbUser, moduleName, timestamp);
+    accessModuleService.updateBypassTime(
+        dbUser.getUserId(), AccessUtils.storageAccessModuleToClient(moduleName), isBypass);
+    if (moduleName == AccessModuleName.PROFILE_CONFIRMATION) {
       // dual-write to DbUser and AccessModuleService
       // we will remove the module fields in DbUser soon
       // see the Access Module Update epic
       // https://precisionmedicineinitiative.atlassian.net/browse/RW-6237
-
       dbUser.setProfileLastConfirmedTime(timestamp);
       dbUser = userDao.save(dbUser);
-      accessModuleService.updateCompletionTime(dbUser, moduleName, timestamp);
-    } else {
-      throw new IllegalArgumentException("Not implemented!");
     }
+
+    final String time = Optional.ofNullable(timestamp).map(Timestamp::toString).orElse("NULL");
+    LOG.info(
+        String.format(
+            "Updating %s completion time for user %s to %s; bypass time to %s",
```

Comment:
Change to: "Updating %s bypass and/or completion time for user %s to %s; bypass mode is %s",